app/actions/ats_client.py

```diff
@@ -0,0 +1,207 @@
+import logging
+import httpx
+import pydantic
+import xmltodict
+import stamina
+
+from datetime import datetime, timedelta
+from xml.parsers.expat import ExpatError
+from typing import List, Optional
+
+
+logger = logging.getLogger(__name__)
+
+
+# Pydantic models
+class DataResponse(pydantic.BaseModel):
+    ats_serial_num: str = pydantic.Field(..., alias="AtsSerialNum")
+    longitude: float = pydantic.Field(None, alias='Longitude', ge=-180.0, le=360.0)
+    latitude: float = pydantic.Field(None, alias='Latitude', ge=-90.0, le=90.0)
+    date_year_and_julian: datetime = pydantic.Field(..., alias="DateYearAndJulian")
+    num_sats: Optional[str] = pydantic.Field(None, alias='NumSats')
+    hdop: Optional[str] = pydantic.Field(None, alias='Hdop')
+    fix_time: Optional[str] = pydantic.Field(None, alias='FixTime')
+    dimension: Optional[str] = pydantic.Field(None, alias='Dimension')
+    activity: Optional[str] = pydantic.Field(None, alias='Activity')
+    temperature: Optional[str] = pydantic.Field(None, alias='Temperature')
+    mortality: Optional[bool] = pydantic.Field(None, alias='Mortality')
+    low_batt_voltage: Optional[bool] = pydantic.Field(None, alias='LowBattVoltage')
+
+    class Config:
+        allow_population_by_field_name = True
+
+    """
+    @validator('DateYearAndJulian')
+    def parse_datetime(cls, v):
+        if not v.tzinfo:
+            return v.replace(tzinfo=timezone.utc)
+        return v
+    """
+
+
+class TransmissionsResponse(pydantic.BaseModel):
+    date_sent: datetime = pydantic.Field(..., alias="DateSent")
+    collar_serial_num: str = pydantic.Field(..., alias="CollarSerialNum")
+    number_fixes: Optional[int] = pydantic.Field(None, alias='NumberFixes')
+    batt_voltage: Optional[float] = pydantic.Field(None, alias='BattVoltage')
+    mortality: Optional[str] = pydantic.Field(None, alias='Mortality')
+    break_off: Optional[str] = pydantic.Field(None, alias='BreakOff')
+    sat_errors: Optional[str] = pydantic.Field(None, alias='SatErrors')
+    year_base: Optional[str] = pydantic.Field(None, alias='YearBase')
+    day_base: Optional[str] = pydantic.Field(None, alias='DayBase')
+    gmt_offset: Optional[int] = pydantic.Field(None, alias='GmtOffset')
+    low_batt_voltage: Optional[bool] = pydantic.Field(None, alias='LowBattVoltage')
+
+    class Config:
+        allow_population_by_field_name = True
+
+    """
+    @validator('DateSent')
+    def parse_datetime(cls, v):
+        if not v.tzinfo:
+            return v.replace(tzinfo=timezone.utc)
+        return v
+    """
+
+
+class PullObservationsDataResponse(pydantic.BaseModel):
+    vehicles: List[DataResponse]
+
+    @pydantic.validator("vehicles", pre=True)
+    def validate_vehicles(cls, val):
+        if isinstance(val, list):
+            return val
+        # val is not a valid list, return an empty list instead
+        return []
```

Comment:
I'm not familiar with this validator (although it is shown as new code in the PR because I renamed the ats client file, but I haven't modified it really). But I suspect it may have been added to avoid some crash when ATS returns no data, do you remember @vgarcia13?